esx_service/utils/auth_data.py

```diff
@@ -561,10 +568,46 @@ def handle_upgrade_1_1_to_1_2(self):
                      """
             sql_script = script.format(DB_MAJOR_VER, DB_MINOR_VER)
             self.conn.executescript(sql_script)
+
+            logging.debug("handle_upgrade_1_1_to_1_2: update vms table Done")
+
+            # update the tenants table to set "default_datastore" to "__VM_DS" if "default_datastore" is ""
+            self.conn.execute("UPDATE OR IGNORE tenants SET default_datastore_url = ?  where default_datastore_url = ?", (auth_data_const.VM_DS_URL, "" ))
+            logging.debug("handle_upgrade_1_1_to_1_2: update default_datastore in tenants table")
+
+            cur = self.conn.execute("SELECT * FROM tenants")
+            result = cur.fetchall()
+
+            # go through tenants table, insert full access privilege to "default_datastore" for each tenant if not present
+            for r in result:
+                id = r['id']
+                ds_url = r['default_datastore_url']
+                privilege = (id, ds_url, 1, 0, 0)
+                self.conn.execute("INSERT OR IGNORE INTO privileges(tenant_id, datastore_url, allow_create, max_volume_size, usage_quota) VALUES (?, ?, ?, ?, ?)",
+                                  privilege)
+            logging.debug("handle_upgrade_1_1_to_1_2: Insert privilege to default_datastore in privileges table")
+
+            cur = self.conn.execute("SELECT * FROM tenants WHERE id = ?",
+                                    (auth_data_const.DEFAULT_TENANT_UUID,)
+                                   )
+
+            result = cur.fetchall()
+            logging.debug("handle_upgrade_1_1_to_1_2: Check DEFAULT tenant exist")
+            if result:
+                # _DEFAULT tenant exists
+                # insert full access privilege to "__ALL_DS" for "_DEFAULT" tenant
```

Comment:
I think the end result is:
1) When default tenant exists, and it was not edited, then open up ALL_DS for it(Remove the access privilege to "_DEFAULT_DS" and insert an full access privilege to "ALL_DS"
2) When default tenant exists, and it was edited (access privilege to _DEFAULT_DS has been edited by user). I am not sure what is the right behavior in this case. I think it would be replace the "datastore_url" field in  existing access privilege to "_DEFAULT_DS"  from "_DEFAULT_DS_URL" to "_ALL_DS_URL".
3) When default tenant does not exist,  do nothing.